### PR TITLE
first commit with two updated files

### PR DESCRIPTION
**Pull request by mdbonner**
pull request

### CODE REVIEW

File: src/main/java/StringManipulation.java [hunk 1]
```diff
@@ -1,33 +1,154 @@
-public class StringManipulation implements StringManipulationInterface {
+import java.util.ArrayList;
```
Comment (ACBxBC) [inline src/main/java/StringManipulation.java:1]: Very weird that I encountered this too but only after the build was semi-successful.

I did not get the same error until I could actually build and have found that some of my test cases failed as well.

Good catch.

File: src/test/java/StringManipulationTest.java [hunk 2]
```diff
@@ -118,32 +218,69 @@ public void testRestoreString1()
...
+    	manipulatedstring.setString("illegalargumentexception");
+        int [] array;
+        array=new int[]{5,4,3,2,1,0};
+        
+        assertThrows(IllegalArgumentException.class, () -> {
+        	manipulatedstring.restoreString(array);
+        });
     }
 
+    // This will test if the method will throw an IndexOutOfBoundsException().
```
Comment (ACBxBC) [inline src/test/java/StringManipulationTest.java:326]: I believe this and the other previous test case is what I accidentally cut off in both of my submissions.

Good to test the out of bounds exception on array.

File: src/test/java/StringManipulationTest.java [hunk 3]
```diff
@@ -118,32 +218,69 @@ public void testRestoreString1()
         assertEquals(restoreString, "rat");
     }
 
+    // This will test the method if it will swap the two words in the string.
```
Comment (ACBxBC) [inline src/test/java/StringManipulationTest.java:263]: Does this array index work for both whitespace character since each len() index is considered a string character?

File: src/test/java/StringManipulationTest.java [hunk 4]
```diff
@@ -53,29 +68,70 @@ public void testRemoveNthCharacter2() {
...
     }
 
+    // This will test the maintainSpacing of the method. The given string has a total of 5 
+    // spaces that are originally in the string. With false passed in for maintainSpacing,
+    // the output string should retain the 5 spaces and remove the characters at positions 3n.
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("0 3  6  9");
+        assertEquals("0     ", manipulatedstring.removeNthCharacter(3, false));
```
Comment (ACBxBC) [inline src/test/java/StringManipulationTest.java:102]: Good test case.

Would this work if the #n space start is changed? 

I.e. 5n or 10n and so on, but may require a nested test for out of bounds I guess. Hope that somewhat makes sense.

**Review by ACBxBC — APPROVED** [review]
Looks good to me. 

Learned a lot from the test case you made and what I may need to update with related test cases in my implementation.

File: src/test/java/StringManipulationTest.java (hunk 3 above)
Comment (mdbonner) [reply src/test/java/StringManipulationTest.java:263]: Alvin, thanks for your comment! Yes, if I understand your comment correctly, this array will work for whitespace characters since whitespaces are considered strings, or substrings at the specific index.

Based on your comment, I've refactored my code to add an additional test that will explicitly test the whitespace characters. Please see my next push update for the additional test that specifically addresses this comment.

File: src/test/java/StringManipulationTest.java [hunk 5]
```diff
@@ -232,6 +232,24 @@ public void testRestoreString2()
         assertEquals(restoreString, "five four");
 
     }
+    
+    // This new test is in response to Alvin Bautista's GitHub code review:
```
Comment (mdbonner) [inline src/test/java/StringManipulationTest.java:236]: @ACBxBC, this is the new test in response to your comment on whitespaces. Check it out and let me know what you think!

File: src/test/java/StringManipulationTest.java (hunk 2 above)
Comment (mdbonner) [reply src/test/java/StringManipulationTest.java:326]: Alvin, interesting that your test cases somehow got cut off, hopefully you get those back in!

I'm curious, I know that the assignment calls for a test on for IllegalArgumentException() and IndexOutOfBoundsException(), do you think there are other possible exceptions that this method could throw? ArrayIndexOutOfBoundsException(), perhaps?

File: src/main/java/StringManipulation.java (hunk 1 above)
Comment (mdbonner) [reply src/main/java/StringManipulation.java:1]: Alvin, so I seemed to have misunderstood my own solution to the Travis build fail issue. The fundamental problem was my mishandling of the commit & add commands while commenting in/out the import statements. So, if I'm not mistaken again, StringManipulation() does in fact need the import statement for ArrayList and StringManipulationTest() should only have the original import statements from the skeleton, no others. 

After making sure this was the case, my build passed without issues (and without my panic), thank goodness.

In your case, I agree that your build had failed because, like you said, some of your test cases did not pass, and not because of import statement issues like me.
